pallets/ad/src/lib.rs

```diff
@@ -370,29 +371,33 @@ pub mod pallet {
 
             ensure!(meta.remain >= value, Error::<T>::InsufficientBalance);
 
-            let kol_meta = Did::<T>::meta(&kol).ok_or(Error::<T>::NotMinted)?;
-            let nft = kol_meta.nft.ok_or(Error::<T>::NotMinted)?;
+            let preferred_nft_id = Nft::<T>::get_preferred(kol).ok_or(Error::<T>::NotMinted)?;
+            let nft_meta = Nft::<T>::get_meta_of(preferred_nft_id).ok_or(Error::<T>::NotMinted)?;
+
+            ensure!(
+                nft_meta.minted,
+                Error::<T>::NotMinted
+            );
```

Comment:
maybe keep in one line
```suggestion
            ensure!(nft_meta.minted, Error::<T>::NotMinted);
```